aten/src/ATen/quantized/Quantizer.cpp

```diff
@@ -111,19 +111,7 @@ inline Tensor new_qtensor(
     QuantizerPtr quantizer) {
   auto memory_format = options.memory_format_opt().value_or(MemoryFormat::Contiguous);
   auto device = options.device();
-  at::Allocator* allocator = nullptr;
-  // TODO: why isn't this just using GetAllocator
-  if (device.is_cuda()) {
-    allocator = at::detail::getCUDAHooks().getCUDADeviceAllocator();
-  } else if (device.is_cpu()) {
-    allocator = at::getCPUAllocator();
-  } else if (device.is_meta()) {
-    allocator = GetAllocator(kMeta);
-  } else if (device.is_privateuseone()) {
-    allocator = GetAllocator(kPrivateUse1);
-  } else {
-    TORCH_INTERNAL_ASSERT(0, "unrecognized device for new_qtensor: ", device);
-  }
+  Allocator* allocator = globalContext().getDeviceAllocator(device);
```

Comment:
Why do you add a new API here instead of using `c10::GetAllocator()` ?

Reply:
> Why do you add a new API here instead of using `c10::GetAllocator()` ?

Hi, I'm trying to let logic relative with select which device method behind `Context.h`, and won't let other logic repeat implement a long if-else statement.

And cuda seems have different logic for get allocator in here, can't replace all with `c10::GetAllocator()` directly.

https://github.com/pytorch/pytorch/blob/69e125a7e9b3ad50d993323f7a9dc2ee3edc6971/aten/src/ATen/cuda/CUDAContext.cpp#L64-L66

Thanks :D

Reply:
@albanD Hi, does this change make sense or we can do it better? :D